modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -150,6 +224,26 @@ public void handleUpdateAll(
         });
     }
 
+    private void executeBatchGc(@Nullable HybridTimestamp newLwm) {
+        if (newLwm != null) {
+            @Nullable HybridTimestamp oldLwm;
+            do {
+                oldLwm = lastRecordedLwm.get();
+
+                if (oldLwm != null && newLwm.compareTo(oldLwm) <= 0) {
+                    break;
+                }
+            } while (!lastRecordedLwm.compareAndSet(oldLwm, newLwm));
+
+            if (oldLwm == null || newLwm.compareTo(oldLwm) > 0) {
+                // Iff the lwm we have is the new lwm.
+                // Otherwise our newLwm is either was smaller than last recorded lwm or last recorded lwm has changed
+                // concurrently and it become greater. If that's the case, another thread will perform the GC.
+                vacuumBatch(newLwm, gcOnUpdateBatchSize);
```

Comment:
The JIRA ticket says:

> To account for that, each update can be preceded with the manual GC of **that** row.

But here, a different approach is implemented (using the queue). Should this be noted in the JIRA issue to make sure that later it is easier to understand what was going on?

Reply:
Yes, it is noted in the comment